### DISCUSSION TITLE
iOS loads a custom filament error

### DISCUSSION THREAD

**Discussion by guoaobao** [google/filament#6742] [Q&A]
iOS loads a custom filament error
![image](https://user-images.githubusercontent.com/17892094/232765618-a3ee6f11-c18e-4781-bd64-0a3b7b3a23ec.png)

**Comment by bejado** [comment]
> Compiler encountered an internal error

Haven't seen that one before. What device / iOS version are you using?

**Reply by guoaobao** [reply]
iphone 7  ,system: ios 11.4

**Reply by bejado** [reply]
Unfortunately that error message is coming from Metal and it doesn't provide any actionable information. Are you able to share your materials here?

**Reply by guoaobao** [reply]
material.roughness = texture(materialParams_roughnessTexture,uv).g;
material.metallic = texture(materialParams_metallicTexture,uv).b;

.mat file will report an error with this statement

**Reply by bejado** [reply]
What's the full material?
